File: string/392. Is_Subsequence.c
bool isSubsequence(char* s, char* t) {
    int start_1, start_2, end_1, end_2;
    start_1 = start_2 = 0;
    end_1 = strlen(s);
    end_2 = strlen(t);
    while((start_1<end_1)&&(start_2<end_2)){
        if(s[start_1]==t[start_2]){start_1++;}
        start_2++;
    }
    return start_1 == end_1;
}

File: string/392. Is_Subsequence.py
def isSubsequence(s: str, t: str) -> bool:
    start_1, end_1 = 0, len(s)
    start_2, end_2 = 0, len(t)
    while(start_2<end_2 and start_1<end_1):
        if s[start_1] == t[start_2]:
            start_1 += 1
        start_2 += 1
    if start_1 == end_1:
            return True
    return False